src/shared/update-avatars.ts

```diff
@@ -0,0 +1,29 @@
+const UPDATE_DELAY = 2000;
+const LOADING_CLASS = 'avatar-loading';
+
+export default function updateAvatars( avatarSelector ) {
+	const images: NodeListOf< HTMLImageElement > = document.querySelectorAll( avatarSelector );
+
+	// Make all the avatars start pulsating
+	images.forEach( ( img ) => {
+		img.classList.add( LOADING_CLASS );
+	} );
+
+	// Wait a bit and then update the URL
+	setTimeout( () => {
```

Comment:
Since `updateAvatars` is a shared function, I suggest canceling the previous `setTimeout` before calling it to avoid potential side effects.